<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Maze</class>
 <widget class="QWidget" name="Maze">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>650</width>
    <height>550</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>650</width>
    <height>550</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>650</width>
    <height>550</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Maze</string>
  </property>
  <widget class="QPushButton" name="pushButton_load_maze">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>10</y>
     <width>121</width>
     <height>41</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>0</width>
     <height>0</height>
    </size>
   </property>
   <property name="focusPolicy">
    <enum>Qt::NoFocus</enum>
   </property>
   <property name="styleSheet">
    <string notr="true">QPushButton {
    border-style: solid;
    border-width: 2px;
    border-radius: 10px;
    border-color: darkGrey;
    font: bold 14px;
    padding: 6px;
}
QPushButton:pressed {
	background-color: rgb(217, 217, 217);
}
</string>
   </property>
   <property name="text">
    <string>Load from file</string>
   </property>
  </widget>
  <widget class="QGroupBox" name="groupBox_prefectMaze">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>60</y>
     <width>120</width>
     <height>231</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <bold>true</bold>
    </font>
   </property>
   <property name="title">
    <string>Perfect maze</string>
   </property>
   <widget class="QPushButton" name="pushButton_save_maze">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>180</y>
      <width>101</width>
      <height>41</height>
     </rect>
    </property>
    <property name="focusPolicy">
     <enum>Qt::NoFocus</enum>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
    border-style: solid;
    border-width: 2px;
    border-radius: 10px;
    border-color: darkGrey;
    font: bold 14px;
    padding: 6px;
}
QPushButton:pressed {
	background-color: rgb(217, 217, 217);
}
</string>
    </property>
    <property name="text">
     <string>Save to file</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_rows">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>30</y>
      <width>58</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>13</pointsize>
      <bold>false</bold>
     </font>
    </property>
    <property name="text">
     <string>Rows</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
    </property>
   </widget>
   <widget class="QSpinBox" name="spinBox_maze_columns">
    <property name="geometry">
     <rect>
      <x>11</x>
      <y>100</y>
      <width>101</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <bold>false</bold>
     </font>
    </property>
    <property name="focusPolicy">
     <enum>Qt::ClickFocus</enum>
    </property>
    <property name="frame">
     <bool>false</bool>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>50</number>
    </property>
    <property name="value">
     <number>25</number>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_generate_maze">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>130</y>
      <width>101</width>
      <height>41</height>
     </rect>
    </property>
    <property name="focusPolicy">
     <enum>Qt::NoFocus</enum>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
    border-style: solid;
    border-width: 2px;
    border-radius: 10px;
    border-color: darkGrey;
    font: bold 14px;
    padding: 6px;
}
QPushButton:pressed {
	background-color: rgb(217, 217, 217);
}
</string>
    </property>
    <property name="text">
     <string>Generate</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="spinBox_maze_rows">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>50</y>
      <width>101</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <bold>false</bold>
     </font>
    </property>
    <property name="focusPolicy">
     <enum>Qt::ClickFocus</enum>
    </property>
    <property name="styleSheet">
     <string notr="true"/>
    </property>
    <property name="frame">
     <bool>false</bool>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>50</number>
    </property>
    <property name="value">
     <number>25</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_columns">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>80</y>
      <width>58</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>13</pointsize>
      <bold>false</bold>
     </font>
    </property>
    <property name="text">
     <string>Columns</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
    </property>
   </widget>
  </widget>
  <widget class="s21::WidgetPainter" name="widget" native="true">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>40</y>
     <width>500</width>
     <height>500</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>500</width>
     <height>500</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>500</width>
     <height>500</height>
    </size>
   </property>
  </widget>
  <widget class="QRadioButton" name="radioButton_change_mode_cave">
   <property name="geometry">
    <rect>
     <x>280</x>
     <y>10</y>
     <width>99</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>Cave mode</string>
   </property>
  </widget>
  <widget class="QRadioButton" name="radioButton_change_mode_maze">
   <property name="geometry">
    <rect>
     <x>140</x>
     <y>10</y>
     <width>99</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>Maze mode</string>
   </property>
  </widget>
  <widget class="QPushButton" name="pushButton_clear">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>500</y>
     <width>121</width>
     <height>41</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>0</width>
     <height>0</height>
    </size>
   </property>
   <property name="focusPolicy">
    <enum>Qt::NoFocus</enum>
   </property>
   <property name="styleSheet">
    <string notr="true">QPushButton {
    border-style: solid;
    border-width: 2px;
    border-radius: 10px;
    border-color: darkGrey;
    font: bold 14px;
    padding: 6px;
}
QPushButton:pressed {
	background-color: rgb(217, 217, 217);
}
</string>
   </property>
   <property name="text">
    <string>Clear all</string>
   </property>
  </widget>
  <widget class="QPushButton" name="pushButton_clear_2">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>450</y>
     <width>121</width>
     <height>41</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>0</width>
     <height>0</height>
    </size>
   </property>
   <property name="focusPolicy">
    <enum>Qt::NoFocus</enum>
   </property>
   <property name="styleSheet">
    <string notr="true">QPushButton {
    border-style: solid;
    border-width: 2px;
    border-radius: 10px;
    border-color: darkGrey;
    font: bold 14px;
    padding: 6px;
}
QPushButton:pressed {
	background-color: rgb(217, 217, 217);
}
</string>
   </property>
   <property name="text">
    <string>Clear path</string>
   </property>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>s21::WidgetPainter</class>
   <extends>QWidget</extends>
   <header>view/widget_painter.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
